### DISCUSSION TITLE
BD no docker (script to create database)

### DISCUSSION THREAD

**Discussion by jmery24** [dotneet/smart-chatbot-ui#58] [Q&A]
Good morning, excellent application, thank you very much for sharing it.
Would it be possible to share the non-dockerized database so that we can see its structure, tables, etc.?
Thank you very much in advance.

**Comment by kendevco** [comment]
I installed mongodb-windows-x86_64-6.0.6-signed.msi on my Windows Server 2019 accepting all defaults and installed Compass when prompted during the install.

Then I used you-know-who to help me write this:

```
import { MongoClient, Db } from 'mongodb';
import crypto from 'crypto';

async function createUser() {
  const client = new MongoClient('mongodb://127.0.0.1:27017/');

  try {
    await client.connect();

    const db: Db = client.db('admin');

    const user = {
      user: "chatui",
      pwd: crypto.randomBytes(16).toString('hex'), // This will generate a random 32-character hexadecimal password
      roles: [
        {
          role: "readWrite",
          db: "chatui"
        }
      ]
    };

    await db.command({ createUser: user.user, pwd: user.pwd, roles: user.roles });

    console.log(`User created successfully. Username: ${user.user}, Password: ${user.pwd}`);
  } catch (err) {
    console.error(err);
  } finally {
    await client.close();
  }
}

createUser();
```

Then in .env or .env.local modify MONGODB entries:

# MongoDB URI for Local
 MONGODB_URI=mongodb://chatui:<password>@127.0.0.1:27017/
 MONGODB_DB=chatui


The database objects are automagically created when they are accessed. This script and connection string are all you should need. Once I got the script in I had to finagle with it for a bit to get the tsx file to run from the debugger. There is a debugger configuration in launch.json for single tsx file I just ran it with that and presto bammo got my database and user. Happy Coding.
